cli/lib/png2src.js

```diff
@@ -130,31 +130,50 @@ function run(sourceFile) {
             const b = png.data[idx + 2];
             const a = png.data[idx + 3];
             const packed = (r << 24) | (g << 16) | (b << 8) | a;
-            let s = palette[packed]
-            if (!s) {
+            if (!palette.has(packed)) {
                 if (colorCount >= 4) {
-                    let RGBs = [];
-                    for (let key in palette) {
-                        RGBs.push(`- (R: ${palette[key].r}, G: ${palette[key].g}, B: ${palette[key].b}, A: ${palette[key].a}); first seen at (${palette[key].x}, ${palette[key].y})`)
-                    }
+                    const rgbHistory = [...palette.values()].map(
+                        c => `- (R: ${c.r}, G: ${c.g}, B: ${c.b}, A: ${c.a}); first seen at (${c.x}, ${c.y})`);
                     throw new Error(`
 Too many colors: maximum is 4. The previous colors were:
-${RGBs.join("\n")}
+${rgbHistory.join("\n")}
 The first occurrence of another color is at (${x}, ${y}) and has the value of (R: ${r}, G: ${g}, B: ${b}, A: ${a})`);
                 }
-                palette[packed] = {
-                    r: r,
-                    g: g,
-                    b: b,
-                    a: a,
+                palette.set(packed, {
+                    r, g, b, a, x, y,
                     i: colorCount++,
-                    x: x,
-                    y: y,
-                };
+                    brightness: (r + g + b) * a,
```

Comment:
Would it be better to use a perceptual brightness here? eg. luma:

```suggestion
                    brightness: (0.2126*r + 0.7152*g + 0.0722*b) * a,
```

Reply:
Sure thing! The "Commit suggestion" button on Github was neat, thank you!